src/amount.js

```diff
@@ -388,7 +388,7 @@ Amount.prototype._copy = function(value) {
 Amount.prototype.compareTo = function(to) {
   var toAmount = Amount.from_json(to);
   if (!this.is_comparable(toAmount)) {
-    return new Amount(NaN);
+    return 0;
```

Comment:
Any return here doesn't sound right. Only right solution is to throw exception, but I don't sure it is acceptable here.